### PR TITLE
[CBRD-23096] Correct multi-insert logging

### PR DESCRIPTION
**Pull request by lucianlazar2211**
http://jira.cubrid.org/browse/CBRD-23096

The system operation was started after the allocation of the heap pages needed for the multi insert. Now, the sysop starts before the pages allocation so that the alloc is added in the current sysop.

### CODE REVIEW

File: src/transaction/locator_sr.c [hunk 1]
```diff
@@ -13826,9 +13829,23 @@ locator_multi_insert_force (THREAD_ENTRY * thread_p, HFID * hfid, OID * class_oi
...
+  if (error_code != NO_ERROR)
+    {
+      // Safeguard
+      ASSERT_ERROR ();
+      log_sysop_abort (thread_p);
+    }
+  else
+    {
+      // Commit the sysop
+      log_sysop_commit (thread_p);
```
Comment (ribram) [inline src/transaction/locator_sr.c]: Actually sysop commit is not right here. Records should be removable in case of rollback/undo.

Comment (eseokoh) [reply src/transaction/locator_sr.c]: it should be `log_sysop_attach_to_outer`.

Comment (ribram) [reply src/transaction/locator_sr.c]: it is actually more complicated than that... insert records should be undone, but undoing heap pages allocations & linking in heap is not so straight-forward.

what happens if a scan is in progress?

Comment (eseokoh) [reply src/transaction/locator_sr.c]: hmm...  We are in a trouble for the case. 

Then, no other option but the current?

Comment (ribram) [reply src/transaction/locator_sr.c]: None of the simple solutions - commit or attach to outer - is satisfying. Unfortunately, I can only think of non-trivial solutions. For instance, ending with a logical undo, that on abort deletes all inserted records.

We may discuss the matter with the whole team and maybe we identify a better solution.

File: src/transaction/locator_sr.c [hunk 2]
```diff
@@ -13821,12 +13821,43 @@ locator_multi_insert_force (THREAD_ENTRY * thread_p, HFID * hfid, OID * class_oi
 	}
     }
 
-  // Now form a heap chain with the pages and add the chain to the current heap.
-  error_code = heap_append_pages_to_heap (thread_p, hfid, *class_oid, heap_pages_array);
-  if (error_code != NO_ERROR)
+  // This append needs to be run on postpone after the commit.
```
Comment (ribram) [inline src/transaction/locator_sr.c]: move logging to a separate function

**Pull request closed without merging**